src/Group.php

```diff
@@ -119,48 +119,63 @@ public function cleanDBonPurge()
-                case 'Group':
-                    /** @var Group $item */
-                    $ong = [];
-                    if ($_SESSION['glpishow_count_on_tabs']) {
-                        $nb = countElementsInTable(
-                            $this->getTable(),
-                            ['groups_id' => $item->getID()]
-                        );
-                    }
-                    $ong[4] = self::createTabEntry(__('Child groups'), $nb);
+            if ($_SESSION['glpishow_count_on_tabs']) {
+                $nb = countElementsInTable($this->getTable(), ['groups_id' => $item->getID()]);
+            }
+            $ong[4] = self::createTabEntry(__('Child groups'), $nb);
+
+            if ($_SESSION['glpishow_count_on_tabs']) {
+                $tree = Session::getSavedOption(__CLASS__, 'tree', 0);
+                $user = Session::getSavedOption(__CLASS__, 'user', 0);
+                $type = Session::getSavedOption(__CLASS__, 'onlytype', '');
+                if ($item->getField('is_itemgroup')) {
+                    $ong[1] = $this->createTabEntryForGroup($item, $CFG_GLPI['linkgroup_types'], 'groups_id', $tree, $user, $type, __('Used items'));
+                }
+                if ($item->getField('is_assign')) {
+                    $ong[2] = $this->createTabEntryForGroup($item, $CFG_GLPI['linkgroup_tech_types'], 'groups_id_tech', $tree, $user, $type, __('Managed items'));
+                }
+            } else {
+                if ($item->getField('is_itemgroup')) {
+                    $ong[1] = __('Used items');
+                }
+                if ($item->getField('is_assign')) {
+                    $ong[2] = __('Managed items');
+                }
+            }
 
-                    if ($item->getField('is_itemgroup')) {
-                        $ong[1] = __('Used items');
-                    }
-                    if ($item->getField('is_assign')) {
-                        $ong[2] = __('Managed items');
-                    }
-                    if (
-                        $item->getField('is_usergroup')
-                        && Group::canUpdate()
-                        && Session::haveRight("user", User::UPDATEAUTHENT)
-                        && AuthLDAP::useAuthLdap()
-                    ) {
-                        $ong[3] = __('LDAP directory link');
-                    }
-                    return $ong;
+            if ($item->getField('is_usergroup') && Group::canUpdate() && Session::haveRight("user", User::UPDATEAUTHENT) && AuthLDAP::useAuthLdap()) {
+                $ong[3] = __('LDAP directory link');
             }
         }
-        return '';
+
+        return $ong;
+    }
+
+    private function createTabEntryForGroup($item, $types, $field, $tree, $user, $type, $label)
+    {
+        if (!in_array($type, $types)) {
+            $type = '';
+        }
+        $datas = [];
+        $nb = $item->getDataItems($types, $field, $tree, $user, 0, $datas);
```

Comment:
This is going to be very slow/inconsistent when there are a lot of items. Even if you ignore the `$datas` array passed by ref, the called method still loads the actual list content into the array rather than just counting the results.

It also would hard-cap the counter to the current list limit even if there are more results.

Reply:
Perhaps simply counting the lines that match the SQL filter would suffice

```patch
diff --git a/src/Group.php b/src/Group.php
index e8617e07bd..ef7b9f8d6e 100644
--- a/src/Group.php
+++ b/src/Group.php
@@ -137,14 +137,20 @@ class Group extends CommonTreeDropdown
             $ong[4] = self::createTabEntry(__('Child groups'), $nb);
 
             if ($_SESSION['glpishow_count_on_tabs']) {
-                $tree = Session::getSavedOption(__CLASS__, 'tree', 0);
-                $user = Session::getSavedOption(__CLASS__, 'user', 0);
-                $type = Session::getSavedOption(__CLASS__, 'onlytype', '');
                 if ($item->getField('is_itemgroup')) {
-                    $ong[1] = $this->createTabEntryForGroup($item, $CFG_GLPI['linkgroup_types'], 'groups_id', $tree, $user, $type, __('Used items'));
+                    $total_linkgroups = 0;
+                    foreach ($CFG_GLPI['linkgroup_types'] as $itemtype_linked) {
+                        $total_linkgroups += countElementsInTable($itemtype_linked::getTable(), ['groups_id' => $item->getID()]);
+                    }
+                    $ong[1] = self::createTabEntry(__('Used items'), $total_linkgroups);
                 }
+
                 if ($item->getField('is_assign')) {
-                    $ong[2] = $this->createTabEntryForGroup($item, $CFG_GLPI['linkgroup_tech_types'], 'groups_id_tech', $tree, $user, $type, __('Managed items'));
+                    $total_linkgroup_tech_types = 0;
+                    foreach ($CFG_GLPI['linkgroup_tech_types'] as $itemtype_linked) {
+                        $total_linkgroup_tech_types += countElementsInTable($itemtype_linked::getTable(), ['groups_id_tech' => $item->getID()]);
+                    }
+                    $ong[2] = self::createTabEntry(__('Managed items'), $total_linkgroup_tech_types);
                 }
             } else {
                 if ($item->getField('is_itemgroup')) {

```

@Lainow  the patch may need to be adapted